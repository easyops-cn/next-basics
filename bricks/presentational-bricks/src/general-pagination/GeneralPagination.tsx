import React from "react";
import { useTranslation } from "react-i18next";
import { NS_PRESENTATIONAL_BRICKS, K } from "../i18n/constants";
import { Pagination } from "antd";
import { PaginationProps } from "antd/lib/pagination";
import style from "./index.module.css";
import classNames from "classnames";

interface GeneralPaginationProps {
  configProps?: PaginationProps;
  onlyShowTotal?: boolean;
  page: number;
  pageSize: number;
  total: number;
  handleOnChange: (page: number, size: number) => void;
}

export function GeneralPagination(
  props: GeneralPaginationProps
): React.ReactElement {
  const { t } = useTranslation(NS_PRESENTATIONAL_BRICKS);

  const showTotal = (totals: number): React.ReactElement => {
    return (
      <span className={style.totalText}>
        {t(K.PAGINATION_TOTAL_TEXT)}{" "}
        <strong className={style.total}>{totals}</strong>{" "}
        {t(K.PAGINATION_TOTAL_UNIT)}
      </span>
    );
  };

  // 默认分页配置
  const defaultPagination: PaginationProps = {
    showTotal,
    current: props.page ?? 1,
    pageSize: props.pageSize ?? 10,
    total: props.total,
    showSizeChanger: true,
    pageSizeOptions: ["10", "20", "50"],
    onChange: props.handleOnChange,
  };

  const configProps = { ...defaultPagination, ...(props.configProps || {}) };

  return (
    <Pagination
      {...configProps}
      className={classNames({
        [style.onlyShowTotal]: props.onlyShowTotal,
      })}
    />
  );
}
