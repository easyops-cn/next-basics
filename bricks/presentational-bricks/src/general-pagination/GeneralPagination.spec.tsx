import React from "react";
import { shallow, mount } from "enzyme";
import { GeneralPagination } from "./GeneralPagination";

describe("GeneralPagination", () => {
  it("should work", () => {
    const handleOnChange = jest.fn();
    const wrapper = shallow(
      <GeneralPagination
        page={1}
        pageSize={20}
        total={50}
        handleOnChange={handleOnChange}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("should handleOnChange to have been called ", () => {
    const handleOnChange = jest.fn();
    const wrapper = mount(
      <GeneralPagination
        page={1}
        pageSize={20}
        total={50}
        handleOnChange={handleOnChange}
      />
    );
    // 选择 第三页
    wrapper.find("Pager").at(2).simulate("click");
    expect(handleOnChange).lastCalledWith(3, 20);
  });
});
