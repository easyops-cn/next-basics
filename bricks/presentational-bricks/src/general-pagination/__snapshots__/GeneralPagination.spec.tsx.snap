// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralPagination should work 1`] = `
<Pagination
  className=""
  current={1}
  onChange={[MockFunction]}
  pageSize={20}
  pageSizeOptions={
    Array [
      "10",
      "20",
      "50",
    ]
  }
  showSizeChanger={true}
  showTotal={[Function]}
  total={50}
/>
`;
